react-app/src/components/property/PropertyDialog.tsx

```diff
@@ -262,44 +263,149 @@ export const PropertyAssessedValueEditDialog = (props: IPropertyAssessedValueEdi
         Id: building.Id,
         Evaluations: building.Evaluations?.map((evalu) => ({
           ...evalu,
-          Value: evalu.Value.replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
+          Value: String(evalu.Value).replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
         })),
       })),
     });
   }, [initialValues, initialRelatedBuildings]);
 
+  const currentYear = new Date().getFullYear();
+  const years = [currentYear];
+  const defaultParcelValues = years.map((year) => ({
+    Year: year,
+    Value: 0,
+    EffectiveDate: Date(),
+    EvaluationKeyId: 0,
+    ParcelId: initialValues?.Id || null,
+  }));
+
+  const evaluationMapToRequest = (
+    evaluations: Partial<ParcelEvaluation>[] | Partial<BuildingEvaluation>[],
+  ) => {
+    return evaluations
+      .filter((evaluation) => evaluation.Value != null)
+      .map((evaluation) => ({
+        ...evaluation,
+        BuildingId: (evaluation as BuildingEvaluation).BuildingId,
+        ParcelId: (evaluation as ParcelEvaluation).ParcelId,
+        Value: parseFloat(String(evaluation.Value)),
+        EvaluationKeyId: 0,
+        Year: evaluation.Year,
+      }));
+  };
+
+  useEffect(() => {
+    if (!initialValues?.Evaluations || initialValues.Evaluations.length === 0) {
+      assessedFormMethods.reset({ Evaluations: defaultParcelValues });
+    } else {
+      assessedFormMethods.reset({
+        Evaluations: initialValues?.Evaluations?.map((evalu) => ({
+          ...evalu,
+          Value: evalu.Value.replace(/[$,]/g, ''), // TODO: Consider some more robust handling for this at the TypeORM level.
+        })),
+        RelatedBuildings: initialRelatedBuildings?.map((building) => ({
+          Id: building.Id,
+          Evaluations: building.Evaluations?.map((evalu) => ({
+            ...evalu,
+            Value: String(evalu.Value).replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
+          })).sort((a, b) => b.Year - a.Year),
+        })),
+      });
+    }
+  }, [initialValues, initialRelatedBuildings]);
```

Comment:
A near-identical useEffect exists on lines 256 to 270. Is it possible to consolidate them?